### PR TITLE
feat: accessibility enhancements in list components

### PR DESCRIPTION
**Pull request by jyash97**

### CODE REVIEW

File: packages/web/src/components/list/MultiDataList.js [hunk 1]
```diff
@@ -435,13 +436,20 @@ class MultiDataList extends Component {
...
+					>
 						{selectAllLabel ? (
 							<li
 								key={selectAllLabel}
 								className={`${
 									this.state.currentValue[selectAllLabel] ? 'active' : ''
 								}`}
+								role="option"
+								aria-checked={!!this.state.currentValue[selectAllLabel]}
+								aria-selected={!!this.state.currentValue[selectAllLabel]}
```
Comment (bietkul) [inline packages/web/src/components/list/MultiDataList.js]: We should avoid code duplication, Can you keep a variable to store `!!this.state.currentValue[selectAllLabel]`, and use it further.

File: packages/web/src/components/list/MultiDataList.js [hunk 2]
```diff
@@ -469,13 +477,16 @@ class MultiDataList extends Component {
 									className={`${
 										this.state.currentValue[item.label] ? 'active' : ''
 									}`}
+									role="option"
+									aria-checked={!!this.state.currentValue[item.label]}
+									aria-selected={!!this.state.currentValue[item.label]}
```
Comment (bietkul) [inline packages/web/src/components/list/MultiDataList.js]: same goes for it.

File: packages/web/src/components/list/MultiDataList.js [hunk 3]
```diff
@@ -469,13 +477,16 @@ class MultiDataList extends Component {
...
 								>
 									<Checkbox
 										className={
 											getClassName(this.props.innerClass, 'checkbox') || null
 										}
 										id={`${this.props.componentId}-${item.label}`}
-										name={this.props.componentId}
+										name={`${this.props.componentId}-${item.label}`}
```
Comment (bietkul) [inline packages/web/src/components/list/MultiDataList.js]: You can use `getter` to compute the values.

File: packages/web/src/components/list/MultiList.js [hunk 4]
```diff
@@ -536,13 +535,20 @@ class MultiList extends Component {
...
+					>
 						{selectAllLabel ? (
 							<li
 								key={selectAllLabel}
 								className={`${
 									this.state.currentValue[selectAllLabel] ? 'active' : ''
 								}`}
+								role="option"
+								aria-checked={!!this.state.currentValue[selectAllLabel]}
+								aria-selected={!!this.state.currentValue[selectAllLabel]}
```
Comment (bietkul) [inline packages/web/src/components/list/MultiList.js]: avoid duplication

File: packages/web/src/components/list/MultiList.js [hunk 5]
```diff
@@ -570,13 +576,16 @@ class MultiList extends Component {
 									className={`${
 										this.state.currentValue[item.key] ? 'active' : ''
 									}`}
+									role="option"
+									aria-checked={!!this.state.currentValue[item.key]}
+									aria-selected={!!this.state.currentValue[item.key]}
```
Comment (bietkul) [inline packages/web/src/components/list/MultiList.js]: avoid duplication

File: packages/web/src/components/list/MultiList.js [hunk 6]
```diff
@@ -570,13 +576,16 @@ class MultiList extends Component {
...
 								>
 									<Checkbox
 										className={
 											getClassName(this.props.innerClass, 'checkbox') || null
 										}
 										id={`${this.props.componentId}-${item.key}`}
-										name={this.props.componentId}
+										name={`${this.props.componentId}-${item.key}`}
```
Comment (bietkul) [inline packages/web/src/components/list/MultiList.js]: Use getter

File: packages/web/src/components/list/SingleDataList.js [hunk 7]
```diff
@@ -354,19 +355,27 @@ class SingleDataList extends Component {
...
+						role="radiogroup"
+					>
 						{selectAllLabel && (
 							<li
 								key={selectAllLabel}
 								className={`${
 									this.state.currentValue === selectAllLabel ? 'active' : ''
 								}`}
+								role="radio"
+								aria-checked={this.state.currentValue === selectAllLabel}
```
Comment (bietkul) [inline packages/web/src/components/list/SingleDataList.js]: You can use getter here too.

**Review by bietkul — CHANGES_REQUESTED** [review]
I can see a lot of code duplication when computing the conditions.
It'll be better if can refactor it out.

**Comment by jyash97** [comment]
Sure I will do these changes.

File: packages/web/src/components/list/MultiDataList.js [hunk 8]
```diff
@@ -424,6 +425,8 @@ class MultiDataList extends Component {
 
 		const listItems = this.listItems;
 
+		const isAllChecked = selectAllLabel ? !!this.state.currentValue[selectAllLabel] : null;
+
```
Comment (bietkul) [inline packages/web/src/components/list/MultiDataList.js:429]: Code like that is always confusing for the reader, why you're assigning `null` to a boolean variable.

File: packages/web/src/components/list/MultiList.js [hunk 9]
```diff
@@ -525,6 +524,8 @@ class MultiList extends Component {
 
 		const listItems = this.listItems;
 
+		const isAllChecked = selectAllLabel ? !!this.state.currentValue[selectAllLabel] : null;
+
```
Comment (bietkul) [inline packages/web/src/components/list/MultiList.js:528]: Same issue is here use `false` instead of `null`

**Review by bietkul — CHANGES_REQUESTED** [review]

**Pull request merged**